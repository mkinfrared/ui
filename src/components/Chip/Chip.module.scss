@import "../../styles/common.scss";

.Chip {
  --background: var(--black-light);
  --border-color: var(--grey-dark);
  --delete-color: var(--primary);
  --icon-color: var(--secondary);

  align-items: center;
  background-color: var(--background);
  border: 1px solid;
  border-color: var(--border-color);
  border-radius: 1.8rem;
  cursor: pointer;
  display: inline-flex;
  font-size: 1.4rem;
  gap: 0.6rem;
  padding: 0.2rem 1rem;
  white-space: nowrap;
  width: fit-content;

  @include theme("light") {
    --background: var(--white-dark);
    --border-color: var(--grey-light);
  }

  .cancel {
    height: 1.6rem;

    svg {
      height: 100%;
    }

    path:last-child {
      fill: var(--icon-color);
      transition: fill 250ms var(--ease-1);
    }

    &:hover {
      --icon-color: var(--secondary-light);
    }
  }
}
