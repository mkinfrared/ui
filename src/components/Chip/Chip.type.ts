export type ChipProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  label?: string;
  onClick?: () => void;
  onDelete?: () => void;
};
