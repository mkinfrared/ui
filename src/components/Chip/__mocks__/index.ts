export { default } from "./Chip.mock";
