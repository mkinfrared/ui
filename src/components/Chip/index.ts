export { default } from "./Chip";

export * from "./Chip.type";
