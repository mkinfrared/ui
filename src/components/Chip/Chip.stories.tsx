import { Meta, Story } from "@storybook/react";

import Chip from "./Chip";
import { ChipProps } from "./Chip.type";
import css from "./Story.module.scss";

export default {
  title: "UI/Chip",
  component: Chip,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<ChipProps> = (args) => (
  <div className={css.Story}>
    <Chip {...args} />
  </div>
);

const Default = Template.bind({});
const WithDelete = Template.bind({});

Default.args = {
  label: "Marklar",
  onDelete: undefined,
};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

WithDelete.args = {
  label: "Deletable",
};

export { Default, WithDelete };
