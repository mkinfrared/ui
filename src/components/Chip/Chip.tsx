import { MouseEvent } from "react";

import FakeButton from "components/FakeButton";
import { ReactComponent as Cancel } from "icons/cancel.svg";
import { classNames } from "utils";

import css from "./Chip.module.scss";
import { ChipProps } from "./Chip.type";

const Chip = ({ className, label, onClick, onDelete }: ChipProps) => {
  const handleDeleteClick = (event: MouseEvent) => {
    event.stopPropagation();

    onDelete?.();
  };

  return (
    <FakeButton
      className={classNames(css.Chip, className)}
      data-testid="Chip"
      onClick={onClick}
    >
      <span>{label}</span>
      {!!onDelete && (
        <FakeButton className={css.cancel} onClick={handleDeleteClick}>
          <Cancel />
        </FakeButton>
      )}
    </FakeButton>
  );
};

export default Chip;
