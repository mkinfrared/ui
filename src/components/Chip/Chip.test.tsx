import { fireEvent, render } from "@testing-library/react";

import Chip from "./Chip";

describe("<Chip />", () => {
  const label = "marklar";
  const onClick = jest.fn();
  const onDelete = jest.fn();

  const Component = (
    <Chip label={label} onClick={onClick} onDelete={onDelete} />
  );

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should be defined", () => {
    expect(Chip).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Chip");

    expect(element).toBeDefined();
  });

  it("should render a delete button when 'onDelete' prop is passed", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Chip");
    const cancelButton = element.querySelector(".cancel");

    expect(cancelButton).not.toBeNull();
  });

  it("should call 'onDelete' from props on cancel button click", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Chip");
    const cancelButton = element.querySelector(".cancel");

    fireEvent.click(cancelButton!);

    expect(onDelete).toHaveBeenCalledTimes(1);
  });
});
