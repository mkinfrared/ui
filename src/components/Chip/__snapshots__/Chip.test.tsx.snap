// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> should match the snapshot 1`] = `
<div>
  <div
    class="FakeButton Chip"
    data-testid="Chip"
    role="button"
    tabindex="0"
  >
    <span>
      marklar
    </span>
    <div
      class="FakeButton cancel"
      data-testid="FakeButton"
      role="button"
      tabindex="0"
    >
      <svg>
        cancel.svg
      </svg>
    </div>
  </div>
</div>
`;
